src/cs50/sql.py

```diff
@@ -120,6 +120,14 @@ def execute(self, sql, *args, **kwargs):
         if len(args) > 0 and len(kwargs) > 0:
             raise RuntimeError("cannot pass both named and positional parameters")
 
+        # Infer command from (unflattened) statement
+        for token in statements[0]:
+            if token.ttype in [sqlparse.tokens.Keyword.DDL, sqlparse.tokens.Keyword.DML]:
+                command = token.value.upper()
+                break
+        else:
+            command = None
+
```

Comment:
I thinks this is equivalent to the following fwiw:

```
command = next((t.value.upper() for t in statements[0]
            if t.ttype in (sqlparse.tokens.Keyword.DDL, sqlparse.tokens.Keyword.DML)), None)
```

Reply:
Just not as readable I think!